test/logger_test.rb

```diff
@@ -0,0 +1,34 @@
+# typed: true
+# frozen_string_literal: true
+
+require_relative "./test_helper"
+
+module ShopifyAPITest
+  class LoggerTest < Minitest::Test
+    def test_call_warn
+      ShopifyAPI::Logger.expects(:send_to_logger).with(:warn, "test")
+      ShopifyAPI::Logger.warn("test")
+    end
+
+    def test_deprecated_should_raise_an_error_if_version_is_lower_or_equal_to_current
+      parent_log_level = ShopifyAPI::Context.log_level
```

Comment:
This would be a great use of stubbing `ShopifyAPI::Context.stubs(:log_level).returns(:warn)`. With that stub you don't have to worry about setting the log level back to level you found it in. Once it exits this method the stub will be ignored by the rest of the test.

I typically only capture current settings and put them back how I found them when working with environment variables since they don't have that same contextual rollback a stub has.

Reply:
Really good point I'll add that in!